lib/webdriveragent.js

```diff
@@ -111,11 +112,17 @@ class WebDriverAgent {
   }
 
   /**
+   * Return bundle id for WebDriverAgent to launch the WDA.
+   * The primary usage is with 'this.usePreinstalledWDA'.
+   * It adds `.xctrunner` as suffix by default but 'this.doNotAddXctrunnerSuffix'
+   * lets skip it.
    *
    * @returns {string} Bundle ID for Xctest.
    */
   get bundleIdForXctest () {
-    return this.updatedWDABundleId ? `${this.updatedWDABundleId}.xctrunner` : WDA_RUNNER_BUNDLE_ID_FOR_XCTEST;
+    return this.updatedWDABundleId ?
+      `${this.updatedWDABundleId}${this.doNotAddXctrunnerSuffix ? '' : '.xctrunner'}`
```

Comment:
I am not quite sure we need to completely skip adding the suffix. There is a convention that xctest apps should have bundle ids ending with `xctrunner`. Perhaps it would be just enough to check if the id already ends with and then skip adding the suffix

Reply:
bundle id can be non- `.xctrunner` when the bundle id in info.plist updated after wda build like downloading packages from https://github.com/appium/WebDriverAgent/releases and signed properly